File: app/src/main/java/com/alexandreobs/testeconcrete/model/data/remote/GitAPI.java
package com.alexandreobs.testeconcrete.model.data.remote;

import com.alexandreobs.testeconcrete.model.pojo.repositories.GitResult;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GitAPI {

    @GET("search/repositories?q=language:Java&sort=stars")

    Observable<GitResult> getALLRepo(

            @Query("page") int pagina
    );
}

File: app/src/main/java/com/joaoibarra/gitapp/api/GitApi.kt
package com.joaoibarra.gitapp.api

import com.joaoibarra.gitapp.api.model.Pull
import com.joaoibarra.gitapp.api.model.Result
import io.reactivex.Observable
import retrofit2.http.*

public interface GitApi {
    @Headers(
            "Accept: application/vnd.github.v3.text-match+json",
            "Content-type:application/json"
    )
    @GET("search/repositories")
    fun searchByQuery(@Query("q") q: String = "language:Java",
                      @Query("sort") sort: String = "stars",
                      @Query("page") page: Int) : Observable<Result>

    @Headers(
            "Accept: application/vnd.github.v3.text-match+json",
            "Content-type:application/json"
    )
    @GET("repos/{owner}/{repo}/pulls")
    fun searchPullByRepository(@Path("owner") owner: String,
                               @Path("repo") repo: String,
                               @Query("page") page: Int) : Observable<List<Pull>>
}